### PR TITLE
Fix browserify issue

### LINKED ISSUE #70: v1.5.0 error with browserify
I try to use numbro with browserify but this part of code is only compatible with nodejs
```js
// Load all languages
        var fs = require('fs'),
            path = require('path');
        var langFiles = fs.readdirSync(path.join(__dirname, 'languages'));
        langFiles.forEach(function (langFile) {
            numbro.language(path.basename(langFile, '.js'), require(path.join(__dirname, 'languages', langFile)));
        });

```

### LINKED ISSUE #71: Incorrectly reporting hasModule = true
Hi!

I am using numbro in the browser, like this:
```javascript
var numbro = require('numbro');
```

But when compiling the source file I get this error on Chrome:
```
Uncaught TypeError: x.readdirSync is not a function
```

Basically the `hasModule` is returning `true` even when is not running in a NodeJS environment.
```
if (hasModule) {
        module.exports = numbro;

        // Load all languages
        var fs = require('fs'),
            path = require('path');
        var langFiles = fs.readdirSync(path.join(__dirname, 'languages'));
        langFiles.forEach(function (langFile) {
            numbro.language(path.basename(langFile, '.js'), require(path.join(__dirname, 'languages', langFile)));
        });
    }
```

Note:

I am using Browserify v10.2.4.
numbro 1.5.0

Thanks!

### LINKED ISSUE #79: browserify attempts to load all language files
The patch in #66 automatically loads all language files for any commonjs scenario, [browserify](https://github.com/substack/node-browserify) included.

While I can't really complain about doing this on the server with Node.js, doing this in the browser adds unnecessary bloat that I can't easily remove.

Possible solutions off the top of my head:
* provide a way to specify an array of languages which defaults to all languages on Node
* create special logic for Node within the commonjs scope

As a side effect, this might help with #70.

### PR DESCRIPTION
**Pull request by BenjaminVanRyseghem**
Fix #70
Fix #71
Fix #79

### CODE REVIEW

File: numbro.js [hunk 1]
```diff
@@ -961,14 +995,6 @@
...
     if (hasModule) {
         module.exports = numbro;
-
-        // Load all languages
-        var fs = require('fs'),
-            path = require('path');
-        var langFiles = fs.readdirSync(path.join(__dirname, 'languages'));
-        langFiles.forEach(function (langFile) {
-            numbro.language(path.basename(langFile, '.js'), require(path.join(__dirname, 'languages', langFile)));
-        });
```
Comment (clayzermk1) [inline numbro.js]: You implied that you did not want to break #62 or revert #66 in our conversation in #79. By removing this code, doesn't that break #62/#66?

File: tests/languages/all.js [hunk 2]
```diff
@@ -3,16 +3,18 @@
...
-      // test that languages were loaded.
-      // this relies on the fact that all.js is run before any of the languages tests
-      test.ok(languages['zh-CN']);
-      test.done();
-    } else {
-      test.done();
+    languages: function(test) {
+        if (typeof module !== 'undefined' && module.exports) {
+            test.expect(1);
+            numbro.loadLanguages('languages');
```
Comment (clayzermk1) [inline tests/languages/all.js]: I believe the point of this test is to ensure that on Node.js all languages are _automatically_ loaded. This line seems to be an explicit load which would defeat the purpose of this test.

**Comment by clayzermk1** [comment]
@stewart42 care to weigh in on this patch with respect to #62?

**Comment by BenjaminVanRyseghem** [comment]
@clayzermk1 I wanted to test first that the new `inNodejsRuntime` works correctly.

If it does, we can put it back here

File: tests/languages/all.js (hunk 2 above)
Comment (alexkwolfe) [inline tests/languages/all.js]: I agree with @clayzermk1.

**Comment by BenjaminVanRyseghem** [comment]
but maybe having it as a separated functions as you suggested is a good idea :smile:

File: tests/languages/all.js (hunk 2 above)
Comment (BenjaminVanRyseghem) [inline tests/languages/all.js]: the test here is more to test that `loadLanguages` does its job

File: numbro.js [hunk 3]
```diff
@@ -771,14 +765,55 @@
...
+        if (defaults && defaults.format) {
+            numbro.defaultFormat(defaults.format);
+        }
+        if (defaults && defaults.currencyFormat) {
+            numbro.defaultCurrencyFormat(defaults.currencyFormat);
+        }
+    }
+
+    function inNodejsRuntime() {
+        return process.browser === undefined;
```
Comment (clayzermk1) [inline numbro.js]: I'm not sure, but I don't think this is universal for client-side CommonJS implementations. It should work in browserify, but no clue about webpack or others. I don't think it is standardized. `process.title === 'node'` seems to be more standard from what I could find.

File: tests/languages/all.js (hunk 2 above)
Comment (clayzermk1) [inline tests/languages/all.js]: In that case, you should remove the `if (typeof module !== 'undefined' && module.exports) {` conditional. Perhaps there should be another test for `loadLanguages` that does not conflict with checking that Node.js automatically loads all languages?

File: numbro.js (hunk 3 above)
Comment (BenjaminVanRyseghem) [inline numbro.js]: ok :smile: 

It's what I found after some googling.
maybe use both ?

File: numbro.js (hunk 1 above)
Comment (BenjaminVanRyseghem) [inline numbro.js]: I put back the automatic load :smile:

File: numbro.js [hunk 4]
```diff
@@ -815,14 +809,55 @@
...
+        if (defaults && defaults.format) {
+            numbro.defaultFormat(defaults.format);
+        }
+        if (defaults && defaults.currencyFormat) {
+            numbro.defaultCurrencyFormat(defaults.currencyFormat);
+        }
+    }
+
+    function inNodejsRuntime() {
+        return process && process.browser === undefined && (process.title === 'node' || process.title === 'grunt');
```
Comment (clayzermk1) [inline numbro.js]: There is an issue with using this line and browserify, `process` is not defined by default with browserify.

Comment (clayzermk1) [inline numbro.js]: Maybe add a `typeof process !== 'undefined'` at the beginning?

Comment (BenjaminVanRyseghem) [inline numbro.js]: if `process` is undefined, `inNodejsRuntime` returns false, or?

Comment (clayzermk1) [inline numbro.js]: Nah, throws a "process is not defined" error.

Comment (BenjaminVanRyseghem) [inline numbro.js]: weird :smile:

I will fix it then :wink:

**Comment by BenjaminVanRyseghem** [comment]
is it a merge candidate now?

**Comment by clayzermk1** [comment]
LGTM for my purposes, no errors during a quick test.

**Comment by BenjaminVanRyseghem** [comment]
done then ;)

**Comment by BenjaminVanRyseghem** [comment]
in numbro 1.5.2

**Pull request merged**